homeassistant/components/vegehub/config_flow.py

```diff
@@ -0,0 +1,260 @@
+                    }
+                ),
+                errors={},
+            )
+
+        # If we already have an IP address, we can just ask the user if they want to continue
+        return self.async_show_form(step_id="user", errors=errors)
+
+    async def async_step_zeroconf(
+        self, discovery_info: zeroconf.ZeroconfServiceInfo
+    ) -> ConfigFlowResult:
+        """Handle zeroconf discovery."""
+
+        # Extract the IP address from the zeroconf discovery info
+        device_ip = discovery_info.host
+
+        # Keep track of which IP addresses have already had their MAC addresses
+        # discovered. This allows us to skip the MAC address retrieval for devices
+        # that don't need it. This stops us from waking up a hub every time we see
+        # it come on-line.
+        have_mac = False
+        if device_ip in ip_dict:
+            have_mac = True
+
+        self._hostname = discovery_info.hostname.removesuffix(".local.")
+        self._config_url = (
+            f"http://{discovery_info.hostname[:-1]}:{discovery_info.port}"
+        )
+        self._properties = discovery_info.properties
+
+        if not have_mac:
+            self._hub = VegeHub(device_ip)
+
+            await self._hub.retrieve_mac_address()
+
+            if len(self._hub.mac_address) <= 0:
+                _LOGGER.error("Failed to get device config from %s", device_ip)
+                return self.async_abort(reason="cannot_connect")
+
+            try:
+                # Check to see if this MAC address is already in the list.
+                entry = list(ip_dict.keys())[
+                    list(ip_dict.values()).index(self._hub.mac_address)
+                ]
+                if entry:
+                    # If it's already in the list, then it is connected to another
+                    # IP address. Remove that entry.
+                    ip_dict.pop(entry)
+            except ValueError:
+                _LOGGER.info("Zeroconf found new device at %s", device_ip)
+
+            # Add a new entry to the list of IP:MAC pairs that we have seen
+            ip_dict[device_ip] = self._hub.mac_address
+        else:
+            self._hub = VegeHub(device_ip, mac_address=ip_dict[device_ip])
+
+        # Check if this device already exists
+        await self.async_set_unique_id(self._hub.mac_address)
+        self._abort_if_unique_id_configured()
+
+        self.context.update(
+            {
+                "title_placeholders": {"host": self._hostname + " (" + device_ip + ")"},
+                "configuration_url": (self._config_url),
+            }
+        )
+
+        # If the device is new, allow the user to continue setup
+        return await self.async_step_user()
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry: ConfigEntry) -> config_entries.OptionsFlow:
+        """Return the options flow handler for this integration."""
+        return VegehubOptionsFlowHandler(config_entry)
```

Comment:
get rid of the OptionFlow for this first PR

Reply:
Commit f22adea8c7c98a3ff89c6a3d177f1f2881c9517a.